modules/base/rendering/renderablemodel.cpp

```diff
@@ -263,13 +267,67 @@ RenderableModel::RenderableModel(const ghoul::Dictionary& dictionary)
     addProperty(_opacity);
     registerUpdateRenderBinFromOpacity();
 
+    if (dictionary.hasKey(KeyForceRenderInvisible)) {
+        _forceRenderInvisible = dictionary.value<bool>(KeyForceRenderInvisible);
 
-    if (dictionary.hasKey(KeyGeometry)) {
-        ghoul::Dictionary dict = dictionary.value<ghoul::Dictionary>(KeyGeometry);
-        for (int i = 1; i <= dict.size(); ++i) {
-            std::string key = std::to_string(i);
-            ghoul::Dictionary geom = dict.value<ghoul::Dictionary>(key);
-            _geometry.push_back(modelgeometry::ModelGeometry::createFromDictionary(geom));
+        if (!_forceRenderInvisible) {
+            // Asset file have specifically said to not render invisible parts,
+            // do not notify in the log if invisible parts are detected and dropped
+            _notifyInvisibleDropped = false;
+        }
+    }
+
+    if (dictionary.hasKey(KeyGeomModelFile)) {
+        std::string file;
+
+        if (dictionary.hasValue<std::string>(KeyGeomModelFile)) {
+            // Handle single file
+            file = absPath(dictionary.value<std::string>(KeyGeomModelFile));
+            _geometry = ghoul::io::ModelReader::ref().loadModel(
+                file,
+                _forceRenderInvisible,
+                _notifyInvisibleDropped
+            );
+        }
+        else if (dictionary.hasValue<ghoul::Dictionary>(KeyGeomModelFile)) {
+            ghoul::Dictionary fileDictionary = dictionary.value<ghoul::Dictionary>(
+                KeyGeomModelFile
+            );
+            std::vector<std::unique_ptr<ghoul::modelgeometry::ModelGeometry>> geometries;
+
+            for (std::string_view k : fileDictionary.keys()) {
+                // Handle each file
+                file = absPath(fileDictionary.value<std::string>(k));
+                geometries.push_back(ghoul::io::ModelReader::ref().loadModel(
+                    file,
+                    _forceRenderInvisible,
+                    _notifyInvisibleDropped
+                ));
+            }
+
+            if (geometries.size() > 0) {
+                ghoul::modelgeometry::ModelGeometry combinedGeometry =
+                std::move(*geometries[0].release());
```

Comment:
I think I would prefer this to be split up into two lines.  The `release` might trigger a `delete` before we can make the `move` of the ModelGeometry.  Would be better to save the ModelGeometry into the `combinedGeometry` first and then do the `release` (if that is even necessary)